modules/core/src/main/scala/Session.scala

```diff
@@ -371,7 +384,10 @@ object Session {
           proto.prepare(command, typer).map(PreparedCommand.fromProto(_))
 
         override def transaction[A]: Resource[F, Transaction[F]] =
-          Transaction.fromSession(this, namer)
+          Transaction.fromSession(this, namer, ReadCommitted, ReadWrite)
```

Comment:
I think the default here needs to map to `BEGIN` with no arguments, because the user may have done `SET SESSION CHARACTERISTICS AS TRANSACTION ...` and we want those defaults to be used, whatever they may be. So I think`Transaction.fromSession` probably wants an `Option[(TransactionIsolationLevel, TransactionAccessMode)]`.

Reply:
Totally forgot about that. Fixed now :)